frontend/src/components/NewIssueModal/NewIssueModal.tsx

```diff
@@ -119,64 +142,125 @@ const NewIssueModal: React.FC<React.PropsWithChildren<NewIssueModalProps>> = ({
+							text_for_attachment: text,
+							time: timestamp || 0,
 							issue_title: issueTitle,
-							issue_team_id: issueTeamId,
-							issue_type_id: issueTypeId || undefined,
-							issue_description: issueDescription,
+							issue_id: linkedIssue.id,
+							issue_url: linkedIssue.url,
+							integrations,
 						},
-						refetchQueries: [namedOperations.Query.GetErrorIssues],
-						awaitRefetchQueries: true,
 					})
 				}
+			} else if (commentType === 'ErrorComment') {
+				if (commentId) {
+					if (mode === 'Create Issue') {
+						await createIssueForErrorComment({
+							variables: {
+								project_id: project_id!,
+								error_url: currentUrl,
+								error_comment_id: commentId,
+								text_for_attachment: text,
+								issue_title: issueTitle,
+								issue_team_id: issueTeamId,
+								issue_type_id: issueTypeId || undefined,
+								issue_description: issueDescription,
+								integrations,
+								author_name: author,
+							},
+						})
+					} else {
+						await linkIssueForErrorComment({
+							variables: {
+								project_id: project_id!,
+								error_url: currentUrl,
+								error_comment_id: commentId,
+								text_for_attachment: text,
+								issue_title: issueTitle,
+								integrations,
+								author_name: author,
+								issue_url: linkedIssue.url,
+								issue_id: linkedIssue.id,
+							},
+						})
+					}
+				} else if (errorSecureId) {
+					if (mode === 'Create Issue') {
+						await createErrorComment({
+							variables: {
+								project_id: project_id!,
+								error_group_secure_id: errorSecureId,
+								text,
+								text_for_email: issueTitle,
+								error_url: currentUrl,
+								tagged_admins: [],
+								tagged_slack_users: [],
+								author_name: author,
+								integrations,
+								issue_title: issueTitle,
+								issue_team_id: issueTeamId,
+								issue_type_id: issueTypeId || undefined,
+								issue_description: issueDescription,
+							},
+							refetchQueries: [
+								namedOperations.Query.GetErrorIssues,
+							],
+							awaitRefetchQueries: true,
+						})
+					} else {
+						await createErrorCommentForExistingIssue({
+							variables: {
+								project_id: project_id!,
+								error_group_secure_id: errorSecureId,
+								text,
+								text_for_email: issueTitle,
+								error_url: currentUrl,
+								tagged_admins: [],
+								tagged_slack_users: [],
+								author_name: author,
+								integrations,
+								issue_title: issueTitle,
+								issue_url: linkedIssue.url,
+								issue_id: linkedIssue.id,
+							},
+							refetchQueries: [
+								namedOperations.Query.GetErrorIssues,
+							],
+							awaitRefetchQueries: true,
+						})
+					}
+				}
 			} else {
 				throw new Error('Invalid Comment Type: ' + commentType)
 			}
 			onClose()
 			form.reset()
-			message.success('New Issue Created!')
+
+			const toastMessage =
+				mode === 'Create Issue' ? 'New Issue Created!' : 'Issue Linked'
```

Comment:
Just for consistency, should we include `!` for both, or remove for both?

Reply:
Yes. I think we should add ! to both

Reply:
This has been resolved now.